crates/wit-parser/src/abi.rs

```diff
@@ -382,6 +382,20 @@ def_instruction! {
             ty: TypeId,
         } : [record.fields.len()] => [1],
 
+        /// Create an `i32` from a handle.
+        HandleLower {
+            handle: &'a Handle,
+            name: &'a str,
+            ty: TypeId,
+        } : [1] => [1],
+
+        /// Create a handle from an `i32`.
+        HandleLift {
+            handle: &'a Handle,
+            name: &'a str,
+            ty: TypeId,
+        } : [1] => [1],
```

Comment:
This seems like it might need to propagate own/borrow in the long run, but I think it's reasonable to fill that in if and when it comes up

Reply:
Can you clarify what you mean?  `handle` will be either a `Handle::Own(_)` or a `Handle::Borrow(_)`, so that should tell the code generator everything it needs to know, right?

Arguably, supplying just `ty` would be sufficient, since the other two fields can be retrieved from the `Resolve`.

Reply:
Oops I'm blind and missed that! What's here is exactly what I was looking for